### PR TITLE
remove unused imports from notebooks

### PR DESCRIPTION
**Pull request by jameslamb**
This PR proposes removing unused imports in the example notebooks.

Unused imports make these notebooks worse in the following ways:

* increase memory footprint
* increase load time
* distract and confuse users
* introduce unnecessary risk of notebook breaking if an import fails (like if `from thing import whatever` fails because `whatever` is removed from `thing`)

I couldn't think of a good way to easily check for these in CI. We could do something gross where all the notebooks are converted to `.py` files and then checked with `pylint` or `flake8` but I think that could take a lot of effort. So for now, this PR just cleans some things up and doesn't introduce tests to prevent them from creeping back in in the future.

### CODE REVIEW

File: examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb
Comment (jameslamb) [inline examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: since these `-scikit` examples are just reading a single CSV file into a pandas dataframe, I don't think we're getting any benefit from using `s3fs`

File: examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb
Comment (jameslamb) [inline examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb:421]: this change is probably from the fact that I wrote this notebook on my laptop, not in a Saturn environment. So I don't think it's a big deal either way, but we should keep it

File: examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb
Comment (jameslamb) [inline examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb:426]: this is not something I did manually. I see these newline / no-newline changes frequently in diffs when I edit notebooks. No idea where they came from. I don't think they're problematic though.

**Comment by jsignell** [comment]
> We could do something gross where all the notebooks are converted to .py files and then checked with pylint or flake8 but I think that could take a lot of effort. 

I imagine the work that @bhperry has done on integration tests is most of the way there. Maybe it'd make sense to add some linting over there.

File: examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb
Comment (jsignell) [reply examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: Agree.

**Comment by bhperry** [comment]
> > We could do something gross where all the notebooks are converted to .py files and then checked with pylint or flake8 but I think that could take a lot of effort.
> 
> I imagine the work that @bhperry has done on integration tests is most of the way there. Maybe it'd make sense to add some linting over there.

Yeah just waiting on the new release builds to get into main before merging in the updates to integration-tests. They are already converting every example notebook to a python file, so we could definitely add linting there.

The only question is do we want to consider a bad lint on the examples as a failure for integration-tests? It's not really the intended purpose of them, but also the examples are a part of the platform so I don't really have a problem with the idea.

File: examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb
Comment (bhperry) [reply examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb:426]: yeah I think my vs-code setup stripped those out when I was making changes here. I'll see if I can find a setting for that so it doesn't happen again.

Comment (jameslamb) [reply examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb:426]: I think you should, for all types of files in all settings, not trim off the final trailing newline...

**Comment by jameslamb** [comment]
In my humble opinion, integration tests are way too late to find out about linting problems and it's preferable to do it in this project's on-every-commit CI to prevent problematic code from being merged in.

I think it would be annoying to be running those integration tests on the way to a release, and only then discover some linting issue that you have to come back and fix in the repo.

**Comment by jsignell** [comment]
> I think it would be annoying to be running those integration tests on the way to a release, and only then discover some linting issue that you have to come back and fix in the repo.

100% agree. 

I was imagining the integration tests running on a cron job and warning about linting issues as they chug along. But you are right that doesn't sound very helpful. Probably was should just steal ideas from the integration test repo if/when we want to automate this.

File: examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb
Comment (bhperry) [reply examples/examples-cpu/prefect/prefect-cloud-scheduled-scoring.ipynb:426]: Yep, agreed. This is the only file type where that happens, need to dig into the settings to find out why.

**Review by jsignell — APPROVED** [review]
Whoops - forgot to hit approve...

File: examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb
Comment (rikturr) [reply examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: s3fs is required for anonymous connections. If a user doesn't have AWS credentials setup, regular `pd.read_csv()` will fail

Comment (jameslamb) [reply examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: that's not true.

Here's an example in an environment with no AWS creds and no `s3fs`:

```shell
docker run -it continuumio/miniconda3 /bin/bash

pip install pandas

python
```

```python
import pandas as pd
df = pd.read_csv(
    "https://s3.amazonaws.com/nyc-tlc/trip+data/yellow_tripdata_2019-01.csv"
)
df.head()
```

image

```python
import s3fs
```

image

Comment (jameslamb) [reply examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: notice that in this PR, I also changed the protocol from `s3://` to `https://`. You probably ran into credentials issues before because of the combination of no creds AND using `s3://`. But every file in S3 is available over HTTPS too, and public ones don't require any authentication :grinning:

Comment (rikturr) [reply examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: 🤯🤯

Comment (rikturr) [reply examples/examples-cpu/nyc-taxi/hyperparameter-scikit.ipynb:61]: TIL. I'll make sure to modify other code we have around that uses `s3://`

**Pull request merged**
